src/main/java/de/tum/in/www1/bamboo/server/parser/strategy/ParserUtils.java

```diff
@@ -0,0 +1,58 @@
+package de.tum.in.www1.bamboo.server.parser.strategy;
+
+import nu.xom.Element;
+
+import java.io.File;
+import java.util.regex.Pattern;
+
+/**
+ * Utility class providing shared functionality for report parsing
+ */
+public class ParserUtils {
+
+    /**
+     * Extracts and parses an attribute to an int. Defaults to 0 if parsing fails.
+     *
+     * @param element Element with attributes
+     * @param attribute Attribute  to extract
+     * @return extracted number
+     */
+    public static int extractInt(Element element, String attribute) {
+        try {
+            return Integer.parseInt(element.getAttributeValue(attribute));
+        } catch (NumberFormatException e) {
+            return 0;
+        }
+    }
+
+    /**
+     * Shorten the path to start after the source directory, which is assumed to be '/assignment/src/'.
+     * Returns the path using unix file separators.
+     * If '/assignment/src/' is not found, the whole path will be returned.
+     *
+     * @param path Absolute path like C:\BambooTest\assignment\src\com\abc\staticCodeAnalysis\App.java
+     * @return path like com/abc/staticCodeAnalysis/App.java]
+     */
+    public static String shortenAndTransformToUnixPath(String path) {
+        if (path == null || path.isEmpty()) {
+            return path;
+        }
+        String packageDelimiter = File.separator + "assignment" + File.separator + "src" + File.separator;
```

Comment:
This is a rather strong dependency to the Artemis folder structure.
I would argue that this plugin should be as independent as possible of Artemis.

I would argue that the plugin always reports the full path. Shortening it depending on the folder structure defined in Artemis (which might change in the future) can then be done on the Artemis side!

Reply:
Removed the shortening of file paths.